File: java/src/main/java/io/fivexer/sdk/model/WorkerMetricsWindow.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * A worker's own recent throughput. Medians are null for a worker with no history, not zero.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerMetricsWindow {
    private String workerId;
    private String since;
    private List<WorkerMetricsWindowDay> days;
    private Long medianWaitMs;
    private Long medianCycleMs;

    public String getWorkerId() { return workerId; }
    public String getSince() { return since; }
    public List<WorkerMetricsWindowDay> getDays() { return days; }
    public Long getMedianWaitMs() { return medianWaitMs; }
    public Long getMedianCycleMs() { return medianCycleMs; }
}

File: php/src/Model/WorkerMetricsWindow.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * A worker's own recent throughput. Medians are null for a worker with no history, not zero.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerMetricsWindow
{
    public function __construct(
        public readonly string $workerId,
        public readonly string $since,
        /** @var list<WorkerMetricsWindowDay> */
        public readonly array $days,
        public readonly ?int $medianWaitMs,
        public readonly ?int $medianCycleMs,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            since: (string) ($data['since'] ?? ''),
            days: Json::parseEach($data, 'days', [WorkerMetricsWindowDay::class, 'fromArray']),
            medianWaitMs: isset($data['medianWaitMs']) ? (int) $data['medianWaitMs'] : null,
            medianCycleMs: isset($data['medianCycleMs']) ? (int) $data['medianCycleMs'] : null,
        );
    }
}
